icechunk/src/format/snapshot.rs

```diff
@@ -349,7 +350,7 @@ impl Snapshot {
 
         let message = builder.create_string(&message);
         let parent_id = parent_id.map(|oid| generated::ObjectId12::new(&oid.0));
-        let flushed_at = Utc::now().timestamp_micros() as u64;
+        let flushed_at = flushed_at.unwrap_or_else(Utc::now).timestamp_micros() as u64;
```

Comment:
so we preserve `written_at` now, even when editing? Am I understanding this correctly?

Reply:
yes, this is the original time at which the commit was done

Reply:
we were violating one of IC invariants: snapshots flushed_at are monotonically descending in `ancestry`